### ISSUE TITLE
Scalar UDF Inlining when referencing remote tables

### ISSUE THREAD

**Issue by sgtwinkler** [MicrosoftDocs/sql-docs#6923]
Reporting a bug when a Scalar UDF references a remote table via a linked server connection.  The properties of the subject function show both properties "Is inlineable" and "Marked for inlining" as True for SQL Server 2019 with CU12 applied.  Calls to the function result in the error "The SELECT permission was denied on the object ...Possible failure reasons: Problems with the query, "ResultSet" property not set correctly, parameters not set correctly, or connection not established correctly" because the UDF should not be inlined when referencing a remote table.

A viable work-around was to add another constraint which was recognized as a non-inlineable characteristic; in my function, adding a line to declare a variable and set it to the result of the getdate() function was sufficient to change both the "Is inlineable" and "Marked for inlining" properties as False and avoid the error.

**Comment by WilliamAntonRohm** [comment]
@sgtwinkler -- Dave, thank you for your feedback.  Are you following a particular docs.microsoft.com article?  I'm looking for the best place to file this bug report and will get back to you.

**Comment by sgtwinkler** [comment]
Yes, it is documented at https://docs.microsoft.com/en-us/sql/relational-databases/user-defined-functions/scalar-udf-inlining?view=sql-server-ver15#inlineable-scalar-udfs-requirements.  One item in the list is "The UDF does not reference remote tables"; this is not the case in SQL Server 2019 CU12.  The following article claims this was fixed in CU11: https://support.microsoft.com/en-us/topic/kb4538581-fix-scalar-udf-inlining-issues-in-sql-server-2019-f52d3759-a8b7-a107-1ab9-7fbee264dd5d.  I verified that, with the reference to a remote table (via a linked server) that the function remains "inlineable" and "marked for inlining" according the the sys.sql_modules table (and more easily viewed in the SSMS properties window for the function), and is unaffected by executing sys.sp_refreshsqlmodule on the function.  A workaround was to add the following line which includes the getdate() function, solely to have the database recognize that the function cannot be inlined:

declare @foo datetime = getdate();

This was effective.  Also, it may be possible that the bug was fixed in CU11 but then a regression error added the problem back in CU12; I have only patched my server to CU12 and will not been able to test on CU11.
